dicomImport/matRad_importDicomRTPlan.m

```diff
@@ -0,0 +1,168 @@
+
+% check which type of Radiation is used
+if isfield(planInfo, 'BeamSequence')
+    BeamParam = 'BeamSequence';
+    ControlParam = 'ControlPointSequence';
+elseif isfield(planInfo, 'IonBeamSequence')
+    BeamParam = 'IonBeamSequence';
+    ControlParam = 'IonControlPointSequence';
+else
+    errordlg('Not supported kind of DICOM file.');    
+end
+
+% use the treatment beams only
+BeamSequence = planInfo.(BeamParam);
+BeamSeqNames = fieldnames(BeamSequence);
+for i = 1:length(BeamSeqNames)
+    currBeamSeq             = BeamSequence.(BeamSeqNames{i});
+    try
+        treatDelType = currBeamSeq.TreatmentDeliveryType;
+        if ~strcmpi(treatDelType,'TREATMENT')
+            BeamSequence = rmfield(BeamSequence,BeamSeqNames{i});
+        end
+    catch
+        warning('Something went wrong while determining the type of the beam.');
+    end
+end
+BeamSeqNames = fieldnames(BeamSequence);
+
+%% get information may change between beams
+% loop over beams
+gantryAngles{length(BeamSeqNames)} = [];
+PatientSupportAngle{length(BeamSeqNames)} = [];
+isoCenter{length(BeamSeqNames)} = [];
+for i = 1:length(BeamSeqNames)   
+    currBeamSeq             = BeamSequence.(BeamSeqNames{i});
+    % parameters not changing are stored in the first ControlPointSequence
+    gantryAngles{i}         = currBeamSeq.(ControlParam).Item_1.GantryAngle;
+    PatientSupportAngle{i}  = currBeamSeq.(ControlParam).Item_1.PatientSupportAngle;
+    isoCenter{i}            = currBeamSeq.(ControlParam).Item_1.IsocenterPosition;
+end
+
+% check wether isocenters are consistent
+if numel(isoCenter) > 1
+    if ~isequal(isoCenter{:})
+       errordlg('Values for isocenter are not consistent.')
+    end
+end
+
+% transform iso. At the moment just this way for HFS
+if ct.dicomInfo.ImageOrientationPatient == [1;0;0;0;1;0]
+    isoCenter = isoCenter{1}' - ct.dicomInfo.ImagePositionPatient' + ...
+                         [ct.resolution.x ct.resolution.y ct.resolution.z];
+else
+    error('This Orientation is not yet supported.');
+end
+
+%% read constant parameters
+% readout charge and mass to set radiationMode to matRad specific name
+radiationMode = planInfo.(BeamParam).Item_1.RadiationType;
+if ~strncmpi(radiationMode,'photons',6)
+    try
+        radiationMass = planInfo.(BeamParam).Item_1.RadiationMassNumber;
+        radiationAtomicNumber = planInfo.(BeamParam).Item_1.RadiationAtomicNumber;
+    catch
+        warning('Could not determine mass and atomic number of the particle');
+    end
+end
+
+if strncmpi(radiationMode,'photons',6)
+    radiationMode = 'photons';
+elseif strncmpi(radiationMode,'proton',6)
+    radiationMode = 'protons';
+elseif (strncmpi(radiationMode,'ion',3) && radiationMass == 12 && radiationAtomicNumber == 6)
+    radiationMode = 'carbon';
+else
+    warning('The given type of radiation may not yet supported');
+end
+
+SAD = NaN;
+% load machine
+% temporarily set to HIT or Generic for developing reasons
+if strcmp(radiationMode,'photons')
```

Comment:
I argue that the distinction with regard to machines only makes sense for internal use. Consequently I would rather leave this out...